packages/liveblocks-react-ui/src/primitives/Composer/index.tsx

```diff
@@ -1287,12 +1479,24 @@ const ComposerForm = forwardRef<HTMLFormElement, ComposerFormProps>(
       event.stopPropagation();
     }, []);
 
+    const toggleMark = useCallback(
+      (mark: ComposerBodyMark) => {
+        toggleEditorMark(editor, mark);
+      },
+      [editor]
+    );
+
+    useOnComposerEditorChangeWithEventSource(editorChangeEventSource, () => {
+      setMarks(getMarks(editor));
+    });
```

Comment:
Related to [this comment](https://github.com/liveblocks/liveblocks/pull/2076#discussion_r1859758326), if we do not want to export a hook just for this use case, we could maybe make this a `useEffect` and mimic the implementation of `useOnComposerEditorChangeWithEventSource` directly here.

Reply:
The Composer's `index.tsx` file is already pretty long and complex so I wanted to avoid adding noise just for subscribing, I simplified this by making [a single and generic `useEventSource` hook](https://github.com/liveblocks/liveblocks/pull/2076/commits/2ade5d5747a9fd3409a9e6051e3b004940104d7f).